### PR TITLE
[xbps-query] Added '-P' for custom formatting, and '-u' for unformatted output

### PR DESCRIPTION
**Pull request by aditya3098**
This is useful for doing things like:
$ xbps-install `xbps-query -R -u -s REGEX`

### CODE REVIEW

File: bin/xbps-query/main.c [hunk 1]
```diff
@@ -66,6 +66,8 @@ usage(bool fail)
 	    " -S --show PKG            Show information for PKG [default mode]\n"
 	    " -s --search PKG          Search for packages by matching PKG, STRING or REGEX\n"
 	    "    --cat=FILE PKG        Print FILE from PKG binpkg to stdout\n"
+	    "    -u --unformatted      Print search output without an formattting or description\n"
```
Comment (Gottox) [inline bin/xbps-query/main.c]: typo `formattting`

**Comment by Gottox** [comment]
I'd rather prefer a `--printf` parameter. @chneukirchen had done some work on that iirc.

**Comment by chneukirchen** [comment]
No, I just proposed it.

Also see xpkg in xtools.

**Comment by aditya3098** [comment]
xpkg is still an external tool. However, the printf option sounds better. But probably both a printf and a unformatted option should be provided, as the unformatted way might be used more frequently.
One more thing I was thinking of would be csv/json/other format of easily parsable output, for example to upgrade a large number of servers using a management script/plugin

File: bin/xbps-query/main.c (hunk 1 above)
Comment (aditya3098) [inline bin/xbps-query/main.c]: Fixed in 3d2add5

**Comment by aditya3098** [comment]
If we implement a printf option, how would we specify the true and false strings for installation status?

**Comment by aditya3098** [comment]
Would something like this work:

xbps-query -R --printf "%i %p %v %d" "true" "false" -s blah

**Comment by aditya3098** [comment]
I'm working on a implementation which prints "+" for not installed and "=" for installed. It can be sed'ed to whatever else is desired. Apologize in advance as my computer died so it will take a day or two

**Comment by aditya3098** [comment]
Should I close the pull request and then send another for the manpage changes or make that one now?

**Comment by aditya3098** [comment]
I pushed the man page update, sorry it took a while

**Comment by aditya3098** [comment]
I fixed an error i created in the manpage. Is there a better way to edit manpages than just the text file? XML seems more readable

**Comment by aditya3098** [comment]
Sorry, I didnt notice that commit.

**Comment by Gottox** [comment]
Sooooooooooo... after 1/2 year... Is this read to merge? @xtraeme

**Comment by roman-neuhauser** [comment]
i'm not a "Void distribution member" but i think this PR needs a bit of care. @aditya3098 can you please squash the useful commits into a single one and drop the merges?  there's a few things that need improvements but it's a bit difficult to find the best place to attach my comments to given how scattered and meandering the PR is.

**Comment by aditya3098** [comment]
Okay, Ill do that. I will actually close this PR and move it into a single PR.

**Comment by aditya3098** [comment]
Okay, I am sorry, I am trying to figure out how to do that...

**Comment by roman-neuhauser** [comment]
hello @aditya3098, no need to close this PR.  you probably want to read up on `git rebase --intereactive`.  i can handle that for you if you prefer / don't mind.

**Comment by roman-neuhauser** [comment]
so this goes on in #214

**Pull request closed without merging**